Boa.Constrictor/WebDriver/Extensions/AbstractWebLocatorListQuestion.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Boa.Constrictor.Screenplay;
+using OpenQA.Selenium;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Boa.Constrictor.WebDriver
+{
+    /// <summary>
+    /// Abstract class for any Web questions that return a collection of strings using a Web locator and a property name.
+    /// </summary>
+    public abstract class AbstractWebLocatorListQuestion : AbstractWebLocatorQuestion<IEnumerable<string>>
+    {
+        #region Constructors
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="locator">The target Web element's locator.</param>
+        public AbstractWebLocatorListQuestion(IWebLocator locator) : base(locator) { }
+
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Method that returns a string from a Web element input.
+        /// </summary>
+        protected abstract Func<IWebElement, string> Retrieval { get; }
+
+        #endregion
+
+        #region Methods
+
+        /// <summary>
+        /// Gets the requested value from each element.
+        /// </summary>
+        /// <param name="actor">The actor.</param>
+        /// <param name="driver">The WebDriver.</param>
+        /// <returns></returns>
+        public override IEnumerable<string> RequestAs(IActor actor, IWebDriver driver)
```

Comment:
I do not like the amount of duplicate code being caused by creating new `AbstractWebLocatorListQuestion` and `AbstractWebPropertyListQuestion` parent classes. Notice that `AbstractWebPropertyListQuestion` is nearly identical to `AbstractWebPropertyQuestion`.

I think it would be better to move this method to a static class under the `Extensions` folder, like this:

```csharp
public static class ElementLists
{
    public static IEnumerable<string> GetValues(IActor actor, IWebDriver driver, Func<IWebElement, string> getValue)
    {
            actor.WaitsUntil(Existence.Of(Locator), IsEqualTo.True());
            var elements = driver.FindElements(Locator.Query);
            var strings = from e in elements select getValue(e);

            // ToList() will avoid lazy evaluation
            return strings.ToList();
    }
}
```

The new "List" interactions could be subclasses of existing parent classes. Their implementation methods would simply call this static method as a utility. For example, here's `TextList`:

```csharp
public override IEnumerable<string> RequestAs(IActor actor, IWebDriver driver) =>
    ElementLists.GetValues(actor, driver, e => e.Text);
```

Remember, inheritance is not the only way to share and reuse code. Oftentimes, inheritance causes significant limitations, like what we saw here.